### PR TITLE
Add code samples for documentation

### LINKED ISSUE #19: Create and fill sample file to display .NET examples in MeiliSearch Documentation
# Code Samples for MeiliSearch documentation

## Introduction 
As MeiliSearch grows so does its SDK's.  More and more SDK's rises from the ground and they all deserve to be as well documented as the core engine itself. 

The most common way for a user to understand MeiliSearch is to go to its official documentation. As of yesterday all examples in the documentation were made with `cURL`. Unfortunately, most of our users do not communicate with MeiliSearch directly with `cURL`. Which forces them to search for the specific references somewhere else (in the readme's, in the sdk's code itself,..). This makes for unnecessary friction. 

## Goal

**We want our documentation to include all SDK's**

As a first step we want all examples to be made using the most possible SDK's. As did [Stripe](https://stripe.com/docs/api/authentication) and [Algolia](https://www.algolia.com/doc/guides/managing-results/refine-results/faceting/#retrieving-facets).

sdk_sample

> examples with curl, javascript and soon enough this SDK too!

To achieve this it is expected from this SDK to create a sample file containing all the code samples needed by the documentation.

These are the steps to follow: 
- Create your sample file
- Fill your sample file
- Add your samples to the documentation

## Create your sample file

The sample file is a yaml file added at the root of each MeiliSearch SDK. 
Sample files are created based on the `sample-template.yaml` file. 

`sample-template` file:
```yaml
get_one_index_1: |-
list_all_indexes_1: |-
create_an_index_1: |-
...
```

This template is accessible publicly [here](https://docs.meilisearch.com/sample-template.yaml) or in the public directory : `.vuepress/public/sample-template.yaml` of the documentation.

The name of the file should be `.code-samples.meilisearch.yaml`

## Fill your sample file

After creating the sample file with the content of the sample template you should have a file containing all the `sampleId`'s. 

```yaml
get_one_index_1: |-
list_all_indexes_1: |-
create_an_index_1: |-
...
```

By the name of the different `sampleId` you should know where that code sample will be added to the documentation. 

For example, `create_an_index_1` is the first example in the [API References of the index creation](https://deploy-preview-356--dreamy-wiles-8fd85b.netlify.app/references/indexes.html#create-an-index).

Using the already existing `cURL` example in the documentation you should see what is expected from that sample. It is very important that you look at the already existing samples in the documentation as it gives you the parameters to use in your sample to match with the rest of the documentation.

Good sample based on documentation:
```yaml
create_an_index_1: |-
  client.createIndex({ uid: 'movies' })
```

Bad sample that does not match with the response.
```yaml
create_an_index_1: |-
  client.createIndex({ uid: 'otherName' })
```

Each sample is expected to be **written in the respective SDK language**.

Javascript example: 
```yaml
get_one_index_1: |-
  client.getIndex('movies').show()
list_all_indexes_1: |-
  client.listIndexes()
create_an_index_1: |-
  client.createIndex({ uid: 'movies' })
  ...
```

The complete `cURL` sample file is available at the root of the documentation repository.
Every other SDK sample file should be available at the root of their respective repository.

### Formatting Exception 

There is **one exception** to the formatting rule. 
When the `sampleId` finishes with `_md` it means it is expected to be written in markdown format. 

JavaScript sample id with _md extension:
yaml-js-example

## Add your samples to the documentation

Once your sample file is filled with the code samples, you will need to create a pull request on the [documentation repository](https://github.com/meilisearch/documentation).

Open the following file in your IDE:
`.vuepress/code-samples/sdks.json`

And add your sample file to the list: 
```json
[
  ...
  {
    "language": "sdk-language",
    "label": "sdk-label",
    "url": "url to yaml file"
  }
]
```

The `language` key expect a [supported language for the code highlight](https://meta.stackexchange.com/a/335336).

The `label` key is the name of the tab. While `label` and `language` could have been the same, it created some conflict (i.e: `bash` and `cURL`).

The `url` is the raw link to access your sample file. It should look like this:
`https://raw.githubusercontent.com/[PROJECT]/[REPO]/.code-samples.meilisearch.yaml`

### PR DESCRIPTION
**Pull request by juchom**
# Pull Request

## Related issue
Fixes #19 

## What does this PR do?
- Add missing code samples for meilisearch documentation
- Fix GetStats method to GetStatsAsync
- Fix a typo

Missing samples will be added when #294 #295 will be implemented

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -4,27 +4,46 @@
...
+  var movie = new[]
+  {
+      new Movie { Id = "287947", Name = "Shazam", Genre = "Action" },
+      new Movie { Id = "329996", Name = "Dumbo", Genre = "Cartoon" }
+  };
+  await index.AddDocumentsAsync(movie);
 add_or_update_documents_1: |-
+  var movie = new[]
+  {
+      new Movie { Id = "287947", Name = "Shazam!", Genre = "Action" },
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
      new Movie { Id = "287947", Title = "Shazam ⚡️", Genres = "comedy" },
```

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -4,27 +4,46 @@
...
+  await client.DeleteIndexAsync("movies");
 get_one_document_1: |-
   await client.Index("movies").GetDocumentAsync<Movie>(25684, new List<string> { "id", "title", "poster", "release_date" });
 get_documents_1: |-
+  await client.Index("movies").GetDocumentsAsync<Movie>();
 add_or_replace_documents_1: |-
+  var movie = new[]
+  {
+      new Movie { Id = "287947", Name = "Shazam", Genre = "Action" },
+      new Movie { Id = "329996", Name = "Dumbo", Genre = "Cartoon" }
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
      new Movie {
            Id = "287947",
            Title = "Shazam",
            Poster = "https://image.tmdb.org/t/p/w1280/xnopI5Xtky18MPhK40cZAGAOVeV.jpg",
            Overview = "A boy is given the ability to become an adult superhero in times of need with a single magic word.",
            ReleaseDate = "2019-03-23"
      },
```

File: .code-samples.meilisearch.yaml [hunk 3]
```diff
@@ -4,27 +4,46 @@
...
 add_or_update_documents_1: |-
+  var movie = new[]
+  {
+      new Movie { Id = "287947", Name = "Shazam!", Genre = "Action" },
+  };
+  await index.UpdateDocumentsAsync(movie);
 delete_all_documents_1: |-
+  await client.Index("movies").DeleteAllDocumentsAsync();
 delete_one_document_1: |-
+  await client.Index("movies").DeleteOneDocumentAsync("287947");
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await client.Index("movies").DeleteOneDocumentAsync("25684");
```

File: .code-samples.meilisearch.yaml [hunk 4]
```diff
@@ -4,27 +4,46 @@
...
 delete_documents_1: |-
+  await client.Index("movies").DeleteDocumentsAsync(new[] { "287947", "329996" });
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await client.Index("movies").DeleteDocumentsAsync(new[] { "23488", "153738", "437035", "363869" });
```

File: .code-samples.meilisearch.yaml [hunk 5]
```diff
@@ -62,24 +81,52 @@ update_settings_1: |-
...
 reset_settings_1: |-
+  await client.Index("movies").ResetSettingsAsync();
 get_synonyms_1: |-
+  await client.Index("movies").GetSynonymsAsync();
 update_synonyms_1: |-
+  var synonyms = new Dictionary<string, IEnumerable<string>>
+  {
+  	{ "wolverine", new string[] {"xmen", "logan"} },
+  	{ "logan", new string[] {"wolverine"} }
+  };
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  	{ "logan", new string[] {"wolverine", "xmen"} },
  	{ "wow", new string[] {"world of warcraft"} }
  };
  };
```

File: .code-samples.meilisearch.yaml [hunk 6]
```diff
@@ -108,11 +182,38 @@ filtering_guide_3: |-
...
+      AttributesToCrop = new[] {"overview"},
+      CropLength = 5
+  };
+  await client.Index("movies").SearchAsync<Movie>("shifu", sq);
 search_parameter_guide_highlight_1: |-
+  var sq = new SearchQuery
+  {
+  	AttributesToHighlight = new[] {"overview"}
+  };
+  await client.Index("movies").SearchAsync<Movie>("shifu", sq);
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await client.Index("movies").SearchAsync<Movie>("winter feast", sq);
```

File: .code-samples.meilisearch.yaml [hunk 7]
```diff
@@ -121,18 +222,79 @@ search_parameter_guide_show_matches_position_1: |-
...
 settings_guide_synonyms_1: |-
+  var newSettings = new Settings
+  {
+      Synonyms = new Dictionary<string, IEnumerable<string>>
+      {
+        { "wolverine", new string[] {"xmen", "logan"} },
+        { "logan", new string[] {"wolverine"} }
+      }
+  };
+  await client.Index("movies").UpdateSettingsAsync(newSettings);
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
        { "sweater", new string[] {"jumper"} },
        { "jumper", new string[] {"sweater"} }
      }
  };
  await client.Index("tops").UpdateSettingsAsync(newSettings);
```

File: .code-samples.meilisearch.yaml [hunk 8]
```diff
@@ -121,18 +222,79 @@ search_parameter_guide_show_matches_position_1: |-
...
 settings_guide_stop_words_1: |-
+  var newSettings = new Settings
+  {
+  	StopWords = new[] {"of", "the", "to"}
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  	StopWords = new[] {"the", "a", "an"}
```

File: .code-samples.meilisearch.yaml [hunk 9]
```diff
@@ -121,18 +222,79 @@ search_parameter_guide_show_matches_position_1: |-
...
+  };
+  await client.Index("jackets").UpdateSettingsAsync(newSettings);
 settings_guide_searchable_1: |-
+  var newSettings = new Settings
+  {
+  	SearchableAttributes = new[]
+  	{
+      "title",
+  	  "overview",
+  	  "genres"
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  	  "title",
  	  "overview",
  	  "genres"
```

File: .code-samples.meilisearch.yaml [hunk 10]
```diff
@@ -121,18 +222,79 @@ search_parameter_guide_show_matches_position_1: |-
...
+  	}
+  };
+  await client.Index("movies").UpdateSettingsAsync(newSettings);
 settings_guide_displayed_1: |-
+  var newSettings = new Settings
+  {
+  	DisplayedAttributes = new[]
+  	{
+      "title",
+  	  "overview",
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  	  "title",
  	  "overview",
```

File: .code-samples.meilisearch.yaml [hunk 11]
```diff
@@ -121,18 +222,79 @@ search_parameter_guide_show_matches_position_1: |-
...
       SortableAttributes = new string[] { "price", "author" },
   };
   await client.Index("meteorites").UpdateSettingsAsync(newSettings);
 documents_guide_add_movie_1: |-
+  var movie = new[]
+  {
+      new Movie { Id = "287947", Name = "Shazam", Genre = "Action" },
+      new Movie { Id = "329996", Name = "Dumbo", Genre = "Cartoon" }
+  };
+  await index.AddDocumentsAsync(movie);
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
      new Movie { MovieId = "123sq178", Title = "Amelie Poulain" }
  };
  await Index("movies").AddDocumentsAsync(movie);
```

File: .code-samples.meilisearch.yaml [hunk 12]
```diff
@@ -280,7 +443,14 @@ faceted_search_facets_1: |-
...
   SearchResult<Movie> movies = await client.Index("movies").SearchAsync<Movie>("Batman", filters);
 faceted_search_walkthrough_filter_1: |-
+  var sq = new SearchQuery
+  {
+  	Filter = "(genre = 'Horror' AND genre = 'Mystery') OR director = 'Jordan Peele'"
+  };
+  await client.Index("movies").SearchAsync<Movie>("thriller", sq);
 add_movies_json_1: |-
+  var jsonDocuments = await File.ReadAllTextAsync("movies.json");
+  await index.AddDocumentsJsonAsync(jsonDocuments);
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await Index("movies").AddDocumentsJsonAsync(jsonDocuments);
```

File: .code-samples.meilisearch.yaml [hunk 13]
```diff
@@ -280,7 +443,14 @@ faceted_search_facets_1: |-
 
   SearchResult<Movie> movies = await client.Index("movies").SearchAsync<Movie>("Batman", filters);
 faceted_search_walkthrough_filter_1: |-
+  var sq = new SearchQuery
+  {
+  	Filter = "(genre = 'Horror' AND genre = 'Mystery') OR director = 'Jordan Peele'"
+  };
+  await client.Index("movies").SearchAsync<Movie>("thriller", sq);
 add_movies_json_1: |-
+  var jsonDocuments = await File.ReadAllTextAsync("movies.json");
```
Comment (alallema) [inline .code-samples.meilisearch.yaml:461]: Shouldn't we import System or System.IO?

File: .code-samples.meilisearch.yaml [hunk 14]
```diff
@@ -295,15 +465,45 @@ landing_getting_started_1: |-
 
   var task = await client.Index("movies").AddDocumentsAsync<Movie>(documents);
 post_dump_1: |-
+  await client.CreateDumpAsync();
 phrase_search_1: |-
+  await client.Index("movies").SearchAsync<Movie>("'african american' horror");
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]: For this one we need to use `"` but I'm not sure which one is better this one:
```c#
  await client.Index("movies").SearchAsync<Movie>("\"african american\" horror");
```
or this one:
```c#
  await client.Index("movies").SearchAsync<Movie>('"african american" horror');
```

File: .code-samples.meilisearch.yaml [hunk 15]
```diff
@@ -4,27 +4,46 @@
...
 create_an_index_1: |-
   TaskInfo task = await client.CreateIndexAsync("movies", "id");
 update_an_index_1: |-
   TaskInfo task = await client.UpdateIndexAsync("movies", "id");
 delete_an_index_1: |-
+  await client.DeleteIndexAsync("movies");
 get_one_document_1: |-
   await client.Index("movies").GetDocumentAsync<Movie>(25684, new List<string> { "id", "title", "poster", "release_date" });
 get_documents_1: |-
+  await client.Index("movies").GetDocumentsAsync<Movie>();
```
Comment (alallema) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await client.Index("movies").GetDocumentsAsync<Movie>(new DocumentsQuery() { Limit = 2 });
```

**Review by alallema — COMMENTED** [review]
Thank you so much for this PR @juchom, this is a particularly tedious job ❤️ 
I made some suggestion and comments 😊 

Thanks again for contributing with Meilisearch :heart:
If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2022/), and you would like to receive some swag from Meilisearch too, please complete [this form](https://meilisearch.typeform.com/to/idt5qugY).

**Comment by juchom** [comment]
This one was a real pain, but it is needed ! :)

Thanks for your big review, I'm going to look at it

File: .code-samples.meilisearch.yaml (hunk 14 above)
Comment (juchom) [reply .code-samples.meilisearch.yaml]: In this case the first one.

Second one won't compile, inside single quote is a char.

**Comment by curquiza** [comment]
Wah @juchom thank you so much for this PR!!!

**Review by alallema — APPROVED** [review]
Thanks a lot @juchom for this PR 🔥 🔥 🔥 
LGTM! ✨

**Comment by alallema** [comment]
bors merge

**Comment by alallema** [comment]
_This message is sent automatically_

Thank you for contributing to Meilisearch. If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2022/), and you would like to receive some gift from Meilisearch too, please complete [this form](https://meilisearch.typeform.com/to/idt5qugY).

**Pull request merged**